src/components/icon/icon.js

```diff
@@ -20,25 +21,55 @@ angular.module('material.components.icon', [
  * @restrict E
  *
  * @description
- * The `<md-icon>` directive is an element useful for SVG icons
+ * The `<md-icon>` directive is an element useful for showing an icon
  *
  * @usage
  * <hljs lang="html">
- *  <md-icon icon="/img/icons/ic_access_time_24px.svg">
- *  </md-icon>
+ *  <md-icon md-icon-key="icon-android" md-icon-size=""></md-icon>
  * </hljs>
  *
  */
-function mdIconDirective() {
+function mdIconDirective($mdIcon) {
   return {
+    scope: {
+      mdIconKey: '@',
+      mdIconSize: '@',
+      mdIconColor: '@'
+    },
     restrict: 'E',
-    template: '<object class="md-icon"></object>',
-    compile: function(element, attr) {
-      var object = angular.element(element[0].children[0]);
-      if(angular.isDefined(attr.icon)) {
-        object.attr('data', attr.icon);
+    replace: true,
+    template: '<span class="icon"></span>',
+    link: function(scope, element, attr) {
+      if ($mdIcon[scope.mdIConKey]) {
+        // TODO: use from config.
+      } else {
+        element.append(angular.element('<i>').addClass(scope.mdIconKey));
       }
+      if(angular.isDefined(attr.mdIconKey)) {
```

Comment:
I agree.  I don't like the verbose md-icon-*, but I thought that was required.  I'm ok with md-alias and md-size.  

Not sure what you mean by "using the require:"^mdIcon" rule."